librocksdb_sys/crocksdb/rocksdb/c.h

```diff
@@ -111,6 +111,14 @@ typedef struct crocksdb_ingestexternalfileoptions_t crocksdb_ingestexternalfileo
 typedef struct crocksdb_sstfilewriter_t   crocksdb_sstfilewriter_t;
 typedef struct crocksdb_ratelimiter_t     crocksdb_ratelimiter_t;
 typedef struct crocksdb_pinnableslice_t   crocksdb_pinnableslice_t;
+typedef struct crocksdb_user_collected_properties_t
+    crocksdb_user_collected_properties_t;
+typedef struct crocksdb_table_properties_collection_t
+    crocksdb_table_properties_collection_t;
+typedef struct crocksdb_table_properties_collector_factory_t
+    crocksdb_table_properties_collector_factory_t;
+typedef struct crocksdb_table_properties_collector_factory_context_t
```

Comment:
We rely on the actual structure of `crocksdb_table_properties_collector_factory_context_t`, it's not an opaque struct anymore. So it should be exposed in headers. Or you can introduce a constructor instead of declaring rust.